tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,232 @@
+    machine:
+
+    .. code-block:: shell
+
+        podman machine init --rootful --disk-size 200 --memory 8192 \
+        --cpus 8 -v /var/tmp/tmt:/var/tmp/tmt -v $HOME:$HOME
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+    _id = str(uuid.uuid4())[:8]
+
+    # build a "modified" container image from the base image with tmt dependencies added
+    def _build_derived_image(self, base_image: str) -> str:
+        self._logger.debug("Building modified container image with necessary tmt packages/config")
+        containerfile_template = Template('''
+            FROM $base_image
+
+            RUN \
+            dnf -y install cloud-init rsync && \
+            ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants && \
+            rm /usr/local -rf && ln -sr /var/usrlocal /usr/local && mkdir -p /var/usrlocal/bin && \
+            dnf clean all
+        ''')
+        containerfile = containerfile_template.substitute({"base_image": base_image})
+        with open(f'{self.workdir}/Containerfile', 'w') as file:
+            file.write(containerfile)
+
+        image_tag = f'localhost/tmtmodified-{self._id}'
+        tmt.utils.Command(
+            "podman", "build", f'{self.workdir}',
+            "-f", f'{self.workdir}/Containerfile',
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+        return image_tag
+
+    # build the "base" or user supplied container image
+    def _build_base_image(self, containerfile: str, workdir: str) -> str:
+        image_tag = f'localhost/tmtbase-{self._id}'
+        self._logger.debug("Building container image")
+        tmt.utils.Command(
+            "podman", "build", workdir,
+            "-f", containerfile,
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+        return image_tag
+
+    # build the bootc disk
+    def _build_bootc_disk(self, containerimage: str, image_builder: str) -> None:
+        self._logger.debug("Building bootc disk image")
+        tmt.utils.Command(
+            "podman", "run", "--rm", "--privileged",
+            "-v", "/var/lib/containers/storage:/var/lib/containers/storage",
+            "--security-opt", "label=type:unconfined_t",
+            "-v", f"{self.workdir}:/output",
+            image_builder, "build",
+            "--type", "qcow2",
+            "--local", containerimage
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+    def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
+        """ Provision the bootc instance """
+        super().go(logger=logger)
+
+        data = BootcData.from_plugin(self)
+        data.set_image(f"file://{self.workdir}/qcow2/disk.qcow2")
+        data.show(verbose=self.verbosity_level, logger=self._logger)
+
+        containerimage = ""
```

Comment:
Please, don't use an empty string to represent "unset" or "undefined" or "undecided" value. It's very hard for linters to distinguish from strings that have value, that are all the same to linters. `None` is a more easier to track and much better when it comes to report undefined/unset/undecided/unknown value is passed where defined/set/decided/known value is expected.

IIUIC, eventually this name will carry a valid value, either from `data.containerimage` or from `self._build_base_image()`, therefore I'd drop this declaration completely, `containerimage = data.containerimage` below should set it well-enough for both runtime and linters to understand it.